app/(main)/markets/AiChat.tsx

```diff
@@ -91,12 +137,20 @@ export const AiChat = ({ id }: AiChatProps) => {
                 </div>
               </div>
               <ScrollArea className="h-[460px] md:h-[536px]">
-                <div className="space-y-4 px-4 pb-24 pt-4">
+                <div className="space-y-4 px-4 pb-32 pt-4">
                   {messages.map(message => (
-                    <Message key={message.content.slice(20)} role={message.role}>
+                    <Message key={message.content?.slice(20)} role={message.role}>
```

Comment:
We're getting repeated `key` values in this map.

```suggestion
                  {messages.map((message, idx) => (
                    <Message
                      key={`${idx}-${message.content?.slice(20)}`}
                      role={message.role}
                    >
```

Reply:
Added the index to make it simpler.